### PR TITLE
Added Gentleman65

### PR DESCRIPTION
**Pull request by ramyac6**
## Description

Added Gentleman65 PCB

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
It looks like you need to rebase this PR.

File: keyboards/gentleman65/config.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x00FA
```
Comment (drashna) [inline keyboards/gentleman65/config.h:24]: While this vender ID doesn't appear to be in use, the low number bothers me. 

(doesn't need to be changed)

File: keyboards/gentleman65/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) { /* First encoder */
+    if (clockwise) {
+      tap_code(KC_AUDIO_VOL_UP);
+    } else {
+      tap_code(KC_AUDIO_VOL_DOWN);
+    }
+  }
+}
```
Comment (drashna) [inline keyboards/gentleman65/keymaps/default/keymap.c:46]:
```suggestion
```

File: keyboards/gentleman65/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) { /* First encoder */
+    if (clockwise) {
+      tap_code(KC_AUDIO_VOL_UP);
+    } else {
+      tap_code(KC_AUDIO_VOL_DOWN);
+    }
+  } 
+}
```
Comment (drashna) [inline keyboards/gentleman65/keymaps/via/keymap.c:62]:
```suggestion
```

File: keyboards/gentleman65/gentleman65.c [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "gentleman65.h"
```
Comment (drashna) [inline keyboards/gentleman65/gentleman65.c:18]: This sets the behavior by default, and allows for user keymaps to over-ride the behavior. 

Also, this allows the encoders to work for the QMK Configurator. 
As well is updated to use the proper callbacks for the encoder functions (they changed to booleans recently)

```suggestion
#include "gentleman65.h"

bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) { /* First encoder */
        if (clockwise) {
            tap_code_delay(KC_AUDIO_VOL_UP, 10);
        } else {
            tap_code_delay(KC_AUDIO_VOL_DOWN, 10);
        }
    }
    return true;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/gentleman65/config.h [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+#define MATRIX_ROW_PINS { D3, D2, D1, F7, F1 }
+#define MATRIX_COL_PINS { D4, D6, D7, B4, B5, B6, C6, D5, C7, F0, B2, B1, B3, B0, B7, D0 }
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+#define RGB_DI_PIN F4
+#define RGBLED_NUM 14
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/gentleman65/config.h:44]: could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/gentleman65/rules.mk [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+#   Atmel DFU    atmel-dfu
```
Comment (drashna) [inline keyboards/gentleman65/rules.mk:5]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/gentleman65/rules.mk [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+RGB_MATRIX_ENABLE = no
+RGB_MATRIX_DRIVER = WS2812
+ENCODER_ENABLE = yes
```
Comment (drashna) [inline keyboards/gentleman65/rules.mk:27]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = no
RGB_MATRIX_DRIVER = WS2812
ENCODER_ENABLE = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/gentleman65/info.json [hunk 8]
```diff
@@ -0,0 +1,13 @@
+{
+    "name": "The Gentleman 65", 
+    "url": "https://jkeys.design/products/gentleman-65", 
+    "maintainer": "theengineermachine", 
+    "width": 16, 
+    "height": 5, 
+    "lighting": "qmk_backlight",
```
Comment (fauxpark) [inline keyboards/gentleman65/info.json:7]:
```suggestion
```

File: keyboards/gentleman65/rules.mk [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+RGB_MATRIX_ENABLE = no
+RGB_MATRIX_DRIVER = WS2812
+ENCODER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/gentleman65/rules.mk:27]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = no
RGB_MATRIX_DRIVER = WS2812
ENCODER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/gentleman65/rules.mk [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/gentleman65/rules.mk:21]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by drashna** [comment]
Oh,: 

> ☒ gentleman65: gentleman65: LAYOUT_all: Number of elements in info.json does not match! info.json:69 != LAYOUT_all:68

This needs to be resolved.

**Comment by ramyac6** [comment]
I believe it's fixed now

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**